pkg/proxy/iptables/proxier.go

```diff
@@ -1442,15 +1456,21 @@ func (proxier *Proxier) syncProxyRules() {
 		}
 	}
 
-	// Drop the packets in INVALID state, which would potentially cause
-	// unexpected connection reset.
-	// https://github.com/kubernetes/kubernetes/issues/74839
-	proxier.filterRules.Write(
-		"-A", string(kubeForwardChain),
-		"-m", "conntrack",
-		"--ctstate", "INVALID",
-		"-j", "DROP",
-	)
+	// On environments with complex network topologies dropping invalid conntrack states
+	// can affect legit scenarios, users can decide to set the nf_conntrack_tcp_be_liberal to solve the
+	// undesired effects of INVALID conntract states and not having to use the iptables rules.
+	// Ref: https://github.com/kubernetes/kubernetes/issues/117924
+	if !proxier.tcpLiberal {
+		// Drop the packets in INVALID state, which would potentially cause
+		// unexpected connection reset.
+		// https://github.com/kubernetes/kubernetes/issues/74839
+		proxier.filterRules.Write(
```

Comment:
The two comments would make more sense merged together. (The first comment kinda assumes you've already read the second...)